trantor/net/inner/TimerQueue.cc

```diff
@@ -270,12 +270,27 @@ void TimerQueue::reset(const std::vector<TimerPtr> &expired,
     loop_->assertInLoopThread();
     for (auto const &timerPtr : expired)
     {
-        if (timerPtr->isRepeat() &&
-            timerIdSet_.find(timerPtr->id()) != timerIdSet_.end())
+        // if (timerPtr->isRepeat() &&
+        //     timerIdSet_.find(timerPtr->id()) != timerIdSet_.end())
+        // {
+        //     timerPtr->restart(now);
+        //     insert(timerPtr);
+        // }
+        if (timerIdSet_.find(timerPtr->id()) != timerIdSet_.end())
         {
-            timerPtr->restart(now);
-            insert(timerPtr);
+            if (timerPtr->isRepeat())
+            {
+                timerPtr->restart(now);
+                insert(timerPtr);
+            }
+            else
+            {
+                // If the timer has been called and not repeat, it should be removed from the set
+                // I guess you forget that 
```

Comment:
Yes, you are right, thanks so much, would u please remove the `I guess ..` line, we can discuss in github instead of in comments of source code :)